src/main/java/uk/gov/hmcts/reform/sscs/services/mapper/CaseDataEventBuilder.java

```diff
@@ -0,0 +1,195 @@
+import uk.gov.hmcts.reform.sscs.models.deserialize.gaps2.AppealCase;
+import uk.gov.hmcts.reform.sscs.models.deserialize.gaps2.MajorStatus;
+import uk.gov.hmcts.reform.sscs.models.deserialize.gaps2.MinorStatus;
+import uk.gov.hmcts.reform.sscs.models.idam.IdamTokens;
+import uk.gov.hmcts.reform.sscs.models.serialize.ccd.CaseData;
+import uk.gov.hmcts.reform.sscs.models.serialize.ccd.Event;
+import uk.gov.hmcts.reform.sscs.models.serialize.ccd.Events;
+import uk.gov.hmcts.reform.sscs.models.serialize.ccd.Hearing;
+import uk.gov.hmcts.reform.sscs.services.ccd.SearchCcdService;
+import uk.gov.hmcts.reform.sscs.services.idam.IdamService;
+import uk.gov.hmcts.reform.sscs.util.CcdUtil;
+
+@Service
+class CaseDataEventBuilder {
+
+    private final SearchCcdService searchCcdService;
+    private final IdamService idamService;
+    private final PostponedEventService<uk.gov.hmcts.reform.sscs.models.deserialize.gaps2.Hearing>
+        postponedEventInferredFromDelta;
+    private final PostponedEventService<Hearing> postponedEventInferredFromCcd;
+
+    @Autowired
+    CaseDataEventBuilder(
+        SearchCcdService searchCcdService, IdamService idamService,
+        PostponedEventService<uk.gov.hmcts.reform.sscs.models.deserialize.gaps2.Hearing>
+            postponedEventInferredFromDelta,
+        PostponedEventService<Hearing> postponedEventInferredFromCcd) {
+        this.searchCcdService = searchCcdService;
+        this.idamService = idamService;
+        this.postponedEventInferredFromDelta = postponedEventInferredFromDelta;
+        this.postponedEventInferredFromCcd = postponedEventInferredFromCcd;
+    }
+
+    List<Events> buildPostponedEvent(AppealCase appealCase) {
+        List<Events> events = buildPostponeEventFromMajorStatus(appealCase);
+        events.addAll(buildPostponedEventFromMinorStatus(appealCase));
+        return events;
+    }
+
+    private List<Events> buildPostponedEventFromMinorStatus(AppealCase appealCase) {
+        if (minorStatusIsNotNullAndIsNotEmpty(appealCase.getMinorStatus())) {
+            return appealCase.getMinorStatus().stream()
+                .filter(minorStatus -> areConditionsToCreatePostponedEventMet(minorStatus.getStatusId(), appealCase))
+                .map(minorStatus -> buildNewPostponedEvent(minorStatus.getDateSet()))
+                .distinct()
+                .collect(Collectors.toList());
+        }
+        return Collections.emptyList();
+    }
+
+    private List<Events> buildPostponeEventFromMajorStatus(AppealCase appealCase) {
+        return appealCase.getMajorStatus().stream()
+            .filter(majorStatus -> majorStatus.getStatusId().equals(GapsEvent.RESPONSE_RECEIVED.getStatus()))
+            .filter(majorStatus -> areConditionsFromMajorStatusToCreatePostponedMet(appealCase, majorStatus))
+            .map(majorStatus -> buildNewPostponedEvent(majorStatus.getDateSet()))
+            .distinct()
+            .collect(Collectors.toList());
+    }
```

Comment:
The current major status should be 18 meaning there should be no Date_Closed element set.